runtime/src/registry.rs

```diff
@@ -116,11 +124,52 @@ decl_module! {
             Self::deposit_event(Event::CheckpointCreated(params.checkpoint_id));
             Ok(())
         }
+
+        #[weight = SimpleDispatchInfo::FreeNormal]
+        pub fn set_checkpoint(
+            origin,
+            params: SetCheckpointParams,
+        ) -> Result {
+            ensure_signed(origin)?;
+
+            let mut ancestor_id = params.new_checkpoint_id;
+            // At the end of this loop, the value of `checkpoint_id` will be
+            // the ID of the first ancestor of the checkpoint in
+            // `params: SetCheckpointParams`.
+            while let Some(cp) = store::Checkpoints::get(ancestor_id) {
+                match cp.parent {
+                    None => break,
+                    Some(cp_id) => ancestor_id = cp_id,
+                }
+            };
+            if Some(ancestor_id) != store::InitialCheckpoints::get(params.project_id.clone()) {
+                return Err("The provided checkpoint ID is not a descendant of the project's first checkpoint.")
+            }
+
+            let opt_project = store::Projects::get(params.project_id);
+            let new_project = match opt_project {
+                None => return Err("The provided project ID is not associated with any project."),
+                Some(prj) =>
+                    Project {
+                        id: prj.id,
+                        description: prj.description,
+                        img_url: prj.img_url,
+                        members: prj.members,
+                        current_cp: params.new_checkpoint_id,
+                    }
+            };
+
+            store::Projects::insert(new_project.clone().id, new_project.clone());
+
+            Self::deposit_event(Event::CheckpointChanged(new_project.id, params.new_checkpoint_id));
+            Ok(())
+        }
     }
 }
 decl_event!(
     pub enum Event {
         ProjectRegistered(ProjectId),
         CheckpointCreated(CheckpointId),
+        CheckpointChanged(ProjectId, CheckpointId),
```

Comment:
Let’s keep the event names consistent
```suggestion
        CheckpointSet(ProjectId, CheckpointId),
```